capstone/src/main/webapp/scripts/poll.js

```diff
@@ -26,16 +27,49 @@ function getPollOptions() {
       const optionsContainer = document.getElementById("options-container");
       optionsContainer.innerHTML = "";
       let maxVotes = getMaxVotes(pollData);
+      topChallenge = pollData["options"][0]["text"];
       pollData["options"].forEach(option => {
         renderOptionElement(option, maxVotes);
       });
       return pollData["votedOptions"];
     })
     .then(votedOptions => {
       handleCheck(votedOptions);
+    })
+    .then(checkWeek);
+}
+
+function checkWeek() {
+  getChallenge();
+  let now = new Date();
+  //Hard coding due date for now
+  let millisTillDueDate =
+    new Date(now.getFullYear(), now.getMonth(), now.getDate(), 21, 8, 0, 0) -
+    now;
+  if (millisTillDueDate >= 0) {
+    setTimeout(updatePoll, millisTillDueDate);
+  }
+}
+
+function getChallenge() {
+  fetch("challenge")
+    .then(response => response.json())
+    .then(challengeData => {
+      const weeklyChallenge = document.getElementById("weekly-challenge");
+      weeklyChallenge.innerText = challengeData["challengeName"];
     });
 }
 
+function updatePoll() {
+  fetch("delete-top-option", { method: "POST" }).then(addChallengeToDb);
+}
+
+function addChallengeToDb() {
+  fetch(`challenge?name=${topChallenge}`, { method: "POST" }).then(
+    getChallenge
```

Comment:
remove `getChallenge` or finish implementing `addChallengeToDb` :)

Reply:
I think it is implemented, rather Lucy should add the "()" to make it more explicit you're calling the function